osu.Game/Skinning/PausableSkinnableSound.cs

```diff
@@ -46,9 +54,7 @@ private void load(ISamplePlaybackDisabler samplePlaybackDisabler)
                 samplePlaybackDisabled.BindValueChanged(disabled =>
                 {
                     if (!RequestedPlaying) return;
-
-                    // let non-looping samples that have already been started play out to completion (sounds better than abruptly cutting off).
-                    if (!Looping) return;
+                    if (!AffectedBySamplePlaybackDisable) return;
```

Comment:
This breaks the `else` case below. It's going to allow storyboard samples to start playing a second time after unpause, even when not looping.

Reply:
So I guess this would mean that storyboard samples should never resume on `ISamplePlaybackDisabler` changes, only ever stop. Which sorta makes the entire `AffectedBySamplePlaybackDisable` thing null and void, and at that point probably the entire body of the `samplePlaybackDisabled.BindValueChanged()` callback should be virtual. Does that sound like a viable proposition?

Reply:
At the moment if a storyboard sample is stopped due to the player pausing it doesn't resume, which was a general issue mentioned in #10285. If we only stop a storyboard sample when the `ISamplePlaybackDisabler` changes, unless I'm misunderstanding what would be needed to resume a sample, wouldn't this logic need to change again in the future since it would always skip any `Play()` calls?

Maybe it's not worth worrying about until resuming a sample is sorted out, but I figured it was worth bringing up.

Reply:
I'm thoroughly confused by your comment. Pretty sure the sample playback disabler's state changes aren't used to trigger any sample playback by themselves, they're only used to disable and re-enable them back again in case of breaks/pauses/skips (if needed). Storyboard samples naturally start on their own via `Update()`, and naturally stop on their own via their lifetime ending/a rewind before their start time.

Reply:
So then the code to resume a storyboard sample's playback would be handled in the `Update()` method and not on `ISamplePlaybackDisabler` changes?

I'd seen this code and assumed that it would have been part of the resume logic once it was complete, but it seems I was wrong there.
```csharp
else
{
    // schedule so we don't start playing a sample which is no longer alive.
    scheduledStart = Schedule(() =>
    {
        if (RequestedPlaying)
            base.Play();
    });
}
```

Reply:
Resuming the sample won't currently work anyway because I don't believe there is currently any way to resume samples exposed in framework that won't restart them. While that's the case I don't think resumption should be a concern at all.

Reply:
Okay, I was thinking along the same line but wasn't 100% sure so I thought I would ask.